djangocms_versioning/admin.py

```diff
@@ -42,16 +45,23 @@ class VersioningChangeListMixin:
     """Mixin used for ChangeList classes of content models."""
 
     def get_queryset(self, request):
-        """Limit the content model queryset to latest versions only."""
+        """Limit the content model queryset to the latest versions only."""
         queryset = super().get_queryset(request)
         versionable = versionables.for_content(queryset.model)
 
-        # TODO: Improve the grouping filters to use anything defined in the
-        #       apps versioning config extra_grouping_fields
-        grouping_filters = {}
-        if 'language' in versionable.extra_grouping_fields:
-            grouping_filters['language'] = get_language_from_request(request)
+        """Check if there is a method "self.get_<field>_from_request" for each extra grouping field.
+         If so call it to retrieve the appropriate filter. If no method is found (except for "language")
+         no filter is applied. For "language" the fallback is versioning's "get_language_frmo_request".
 
+         Admins requiring extra grouping field beside "language" need to implement the "get_<field>_from_request"
+         method themselves. A common way to select the field might be GET or POST parameters or user-related settings.
+         """
+        grouping_filters = {}
+        for field in versionable.extra_grouping_fields:
+            if hasattr(self, f"get_{field}_from_request"):
```

Comment:
I'm not seeing any test coverage for this. Is it covered?

Reply:
@Aiky30 Thanks for insisting on this! It turns out to be a bug. The `get_{field}_from_request` method should be looked for in the model admin not the change list!  I added the test (and it turned up the bug)!